### PR TITLE
Add inletsctl to get command

### PR DESCRIPTION
**Pull request by tuananh**
Signed-off-by: Tuan Anh Tran <me@tuananh.org>

Add `inletsctl` to `arkade get` command

## Description

Add `inletsctl` to `arkade get` command

## Motivation and Context

Add `inletsctl` to `arkade get` command

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Tested on macOS

image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by tuananh** [comment]
@alexellis updated as per comment

`inletsctl` needs to use `URLTemplate` and `BinaryTemplate` as well.

eg: url https://github.com/inlets/inletsctl/releases/download/0.5.4/inletsctl-darwin.tgz
extracted file name `inletsctl-darwin` | `inletsctl` | `inletsctl.exe`

image

**Comment by alexellis** [comment]
Sorry @tuananh I just merged another PR and this needs another rebase

Alex

**Comment by tuananh** [comment]
@alexellis rebased.

File: pkg/get/get.go [hunk 1]
```diff
@@ -252,6 +278,41 @@ https://github.com/bitnami-labs/sealed-secrets/releases/download/{{.Version}}/ku
...
+{{$fileName = "inletsctl-arm64.tgz"}}
+{{ else if HasPrefix .OS "ming" -}}
+{{$fileName = "inletsctl.exe.tgz"}}
+{{- else if eq .OS "linux" -}}
+{{$fileName = "inletsctl.tgz"}}
+{{- else if eq .OS "darwin" -}}
+{{$fileName = "inletsctl-darwin.tgz"}}
+{{- end -}}
+https://github.com/inlets/inletsctl/releases/download/{{.Version}}/{{$fileName}}`,
+			BinaryTemplate: `{{ if HasPrefix .OS "ming" -}}
```
Comment (alexellis) [inline pkg/get/get.go:304]: I'm not sure I understand the idea with `BinaryTemplate`? We shouldn't write a file called TOOLNAME-armhf it should always be TOOLNAME, but on Windows, sometimes it needs a .exe suffix.

Comment (tuananh) [reply pkg/get/get.go:304]: @alexellis In this case, the download url is a `tgz` file and the binary file in the `tgz` archive has different name based on arch and platform so I don't know what to pull out from the archive without using `BinaryTemplate` or a new field. 

i saw that the struct already has a field called `BinaryTemplate`, though for different purpose (github release download url with binary file, not compressed) but the name is quite close for the purpose here as well so I reused this.

Comment (alexellis) [reply pkg/get/get.go:304]: Thank you for explaining - this helps. 

So you are looking for what file to copy from the downloaded and extracted archive?
Or are you looking for the final destination name?

We probably need the following, right?

1) Download `inletsctl-darwin.tgz"`
2) Extract `inletsctl-darwin`
3) Copy `inletsctl-darwin` to destination folder as: `inletsctl`

Comment (tuananh) [reply pkg/get/get.go:304]: Im looking for what file to copy from the extracted folder.

**Pull request closed without merging**